buildenv/jenkins/jobs/infrastructure/copyrightCheckDir.sh

```diff
@@ -0,0 +1,612 @@
+#
+# You should have received a copy of the GNU General Public License version
+# 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+#
+# ===========================================================================
+
+trace () {
+  if [ "$VERBOSE" = "1" ]; then
+    echo $1
+  fi
+  return
+}
+
+log () {
+  echo $1 1>&2
+  return
+}
+
+check () {
+  if [ ! "$ROOTDIR" = "" ]; then
+    case "$1" in
+      $ROOTDIR/*) CHECK=1;;
+      *) return 0;;
+    esac
+  fi
+  trace "Checking file $1"
+
+  ERROR=0
+
+  CHECK=0
+  case "$1" in
+     # Comment out to check the test directory files.
+    test/*)
+      trace "Ignoring $1 because it appears to match case parameter expansion test/*"
+      CHECK=0;;
+    # Ignore binary files
+    *.gif)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.gif"
+      CHECK=0;;
+    *.ini)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.ini"
+      CHECK=0;;
+    *.jpg)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.jpg"
+      CHECK=0;;
+    *.jpeg)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.jpeg"
+      CHECK=0;;
+    *.ico)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.ico"
+      CHECK=0;;
+    *.bmp)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.bmp"
+      CHECK=0;;
+    *.png)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.png"
+      CHECK=0;;
+    *.wav)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.wav"
+      CHECK=0;;
+    *.md)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.md"
+      CHECK=0;;
+    *.icu)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.icu"
+      CHECK=0;;
+    *.so)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.so"
+      CHECK=0;;
+    *.dll)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.dll"
+      CHECK=0;;
+    *.exe)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.exe"
+      CHECK=0;;
+    *.bin)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.bin"
+      CHECK=0;;
+    *.zip)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.zip"
+      CHECK=0;;
+    *.jar)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.jar"
+      CHECK=0;;
+    *.class)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.class"
+      CHECK=0;;
+    *.cer)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.cer"
+      CHECK=0;;
+    *.ser)
+      trace "Ignoring $1 because it appears to match case parameter expansion *.ser"
+      CHECK=0;;
+    *) CHECK=1;;
+  esac
+
+  if [ "$CHECK" = '1' ]; then
```

Comment:
This `if` is really long. Could you reverse it to make it cleaner?
```
if [ "$CHECK" = '0' ]; then
    return 0
fi

MAX_LINES=400
```

Reply:
Changed.